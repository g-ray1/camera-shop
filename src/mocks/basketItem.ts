import { cameras } from './cameras';

export const basketItem = {
  camera: cameras[0],
  count: 1,
};
