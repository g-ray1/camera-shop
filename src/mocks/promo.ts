export const promoProduct = {
  id: 1,
  name: 'Ретрокамера Dus Auge lV',
  previewImg: 'img/content/das-auge.jpg',
  previewImg2x: 'img/content/das-auge@2x.jpg',
  previewImgWebp: 'img/content/das-auge.webp',
  previewImgWebp2x: 'img/content/das-auge@2x.webp',
};
