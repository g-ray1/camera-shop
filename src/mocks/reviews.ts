export const reviews = [
  {
    id: '8cc7f38f-eb51-4c35-bdff-acd0dc6f1b63',
    userName: 'Павел',
    advantage: 'Нет.',
    disadvantage: 'Не удобный интерфейс и кнопки',
    review: 'Хорошая камера. Лучше за эти деньги не найти.',
    rating: 4,
    createAt: '2022-11-28T21:00:05.917Z',
    cameraId: 1
  },
  {
    id: '0e8c9a91-7d77-4cd1-9393-4cd511464caa',
    userName: 'Александр',
    advantage: 'Хороший стабилизатор и матрица',
    disadvantage: 'Пришла поврежденная упаковка. Нет теперь понимая со внутренностями',
    review: 'Не возможно найти дополнительные аккамуляторы. К сожалению, те, что идут в комплекте не держут более 7 часов?',
    rating: 2,
    createAt: '2023-02-20T21:00:05.923Z',
    cameraId: 1
  },
  {
    id: '9cefb075-7762-43c7-ab2e-f822e8d363f3',
    userName: 'Александр',
    advantage: 'Рекомендую данный аппарат',
    disadvantage: 'Плохая камера, не рекомендую',
    review: 'В целом для домашнего использования в самый раз!',
    rating: 3,
    createAt: '2022-08-24T21:00:05.923Z',
    cameraId: 1
  },
  {
    id: 'eb8c8add-1df9-4041-8c26-2d6e481ed052',
    userName: 'Дарья',
    advantage: 'Недорогая, за такую цену отличный вариант.',
    disadvantage: 'Тяжелая. Рука быстро устаёт',
    review: 'В целом для домашнего использования в самый раз!',
    rating: 1,
    createAt: '2022-11-22T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: '1224148c-0c1e-454d-a3cd-2d528ffb0658',
    userName: 'Ксения',
    advantage: 'Цена соответствует качеству.',
    disadvantage: 'Быстро садиться зарядка',
    review: 'Отличная камера, мне понравилась, пользуюсь около года. Профессионально оценить нет возможности. Есть разные мелкие недочеты, но, по скольку я не являюсь специалисто в данной области, они не ощущаются в моих работах.',
    rating: 2,
    createAt: '2022-11-25T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: '5d794f96-797f-4630-9a8f-8a1a369e1ec0',
    userName: 'Кирилл',
    advantage: 'Рекомендую данный аппарат',
    disadvantage: 'Плохая камера, не рекомендую',
    review: 'Отличная камера, мне понравилась, пользуюсь около года. Профессионально оценить нет возможности. Есть разные мелкие недочеты, но, по скольку я не являюсь специалисто в данной области, они не ощущаются в моих работах.',
    rating: 1,
    createAt: '2023-02-25T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: 'f0237ea7-c677-45e5-b4c4-c88e178ae54c',
    userName: 'Никита',
    advantage: 'Смущает цена',
    disadvantage: 'Тяжелая. Рука быстро устаёт',
    review: 'Приобрела камеру ориентируясь на отзывы, но выявила множество несоответствий. Цена не оправдала ожидания',
    rating: 3,
    createAt: '2022-08-28T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: 'c2e17a40-a2c3-4013-8346-e6fa44ad293a',
    userName: 'Анастасия',
    advantage: 'Цена соответствует качеству.',
    disadvantage: 'Без объектива',
    review: 'Это моя первая камера. Я в восторге, нареканий нет',
    rating: 5,
    createAt: '2023-01-30T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: '861c8676-3760-4138-982b-6680349d51e0',
    userName: 'Никита',
    advantage: 'Покупали недавно. Пока нареканий нет.',
    disadvantage: 'Пришла поврежденная упаковка. Нет теперь понимая со внутренностями',
    review: 'Приобрела камеру ориентируясь на отзывы, но выявила множество несоответствий. Цена не оправдала ожидания',
    rating: 4,
    createAt: '2023-01-29T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: 'd3fe1116-b138-483f-a163-0baab0cd276b',
    userName: 'Дарья',
    advantage: 'Рекомендую данный аппарат',
    disadvantage: 'Странные звуки при переключении режимов',
    review: 'Это моя первая камера. Я в восторге, нареканий нет',
    rating: 4,
    createAt: '2022-08-30T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: 'f410e901-9565-47dd-aecd-f46770ebaeb5',
    userName: 'Кирилл',
    advantage: 'Смущает цена',
    disadvantage: 'Тяжелая. Рука быстро устаёт',
    review: 'Камера достаточно хорошая, но не очень радует цена.',
    rating: 3,
    createAt: '2022-11-30T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: 'a715856c-9c1c-4da7-a2d9-3d2eb9640953',
    userName: 'Дарья',
    advantage: 'Легкая в плане веса, удобная в интерфейсе, зарядка',
    disadvantage: 'Плохая камера, не рекомендую',
    review: 'Отличная камера. Великолепные снимки, проста в управлении. Полностью оправдывает стоимость',
    rating: 4,
    createAt: '2023-01-29T21:00:05.924Z',
    cameraId: 1
  },
  {
    id: '56fb2343-d3cd-4bf3-827c-6d5ee2103d10',
    userName: 'Александр',
    advantage: 'Цена соответствует качеству.',
    disadvantage: 'Без объектива',
    review: 'Камера достаточно хорошая, но не очень радует цена.',
    rating: 4,
    createAt: '2023-01-22T21:00:05.924Z',
    cameraId: 1
  }
];
