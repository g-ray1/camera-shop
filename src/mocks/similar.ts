import { Camera } from '../types/types';

export const similarCameras: Camera[] = [
  {
    id: 5,
    name: 'Van Shot',
    vendorCode: 'YU7RT5GH76',
    type: 'Коллекционная',
    category: 'Видеокамера',
    description: 'Крайне редкое наименование не потеряло актуальность не смотря на сможество альтернатив. После съёмок на данную камеру фильм не стыдно показать в рамках кинофестиваля. Первые 4К настройки, высочайшее разрешение, уникальная цветопередача.',
    previewImg: 'img/content/van-shot.jpg',
    level: 'Профессиональный',
    rating: 5,
    price: 149990,
    previewImg2x: 'img/content/van-shot@2x.jpg',
    previewImgWebp: 'img/content/van-shot.webp',
    previewImgWebp2x: 'img/content/van-shot@2x.webp',
    reviewCount: 15
  },
  {
    id: 11,
    name: 'SP 520',
    vendorCode: 'JQ756',
    type: 'Коллекционная',
    category: 'Видеокамера',
    description: 'Делайте лучшие фильмы в высоком разрешении. Лёгкая в управлении, мощная начинка, реалистичная цветопередача, возможность просмотра отснятого материала через поворотный ЖК-экран и передача видео через систему Bluetooth.',
    previewImg: 'img/content/sp-520.jpg',
    level: 'Профессиональный',
    rating: 5,
    price: 105590,
    previewImg2x: 'img/content/sp-520@2x.jpg',
    previewImgWebp: 'img/content/sp-520.webp',
    previewImgWebp2x: 'img/content/sp-520@2x.webp',
    reviewCount: 14
  },
  {
    id: 33,
    name: 'Amazing Go',
    vendorCode: 'AD345J',
    type: 'Коллекционная',
    category: 'Видеокамера',
    description: 'Прикрепить камеру на шлем, сделать множество крутых видео под водой или записать свой полёт на тарзанке - благодаря усовершенствованной камере семейства Go ваши приключения останутся навсегда в памяти.',
    previewImg: 'img/content/amazing-go.jpg',
    level: 'Профессиональный',
    rating: 4,
    price: 79000,
    previewImg2x: 'img/content/amazing-go@2x.jpg',
    previewImgWebp: 'img/content/amazing-go.webp',
    previewImgWebp2x: 'img/content/amazing-go@2x.webp',
    reviewCount: 12
  },
  {
    id: 17,
    name: 'Look Identify',
    vendorCode: 'LD2000',
    type: 'Коллекционная',
    category: 'Фотоаппарат',
    description: 'Среднеформатная 40-мегапиксельная камера обладет уникальным непоторимым дизайном и ручной росписью корпуса. Ориентирована на студийную съёмку, имеет полный кадр --  35-мм сенсор ',
    previewImg: 'img/content/look-identify.jpg',
    level: 'Любительский',
    rating: 5,
    price: 126000,
    previewImg2x: 'img/content/look-identify@2x.jpg',
    previewImgWebp: 'img/content/look-identify.webp',
    previewImgWebp2x: 'img/content/look-identify@2x.webp',
    reviewCount: 18
  },
  {
    id: 12,
    name: 'Look Shot',
    vendorCode: 'NB569SH',
    type: 'Цифровая',
    category: 'Видеокамера',
    description: 'Гибридный автофокус и динамический стабилизатор по приятной цене. Светочувствительная матрица  без шумов. 8-кратный выездной зум в стильной упаковке.',
    previewImg: 'img/content/look-shot.jpg',
    level: 'Любительский',
    rating: 4,
    price: 18590,
    previewImg2x: 'img/content/look-shot@2x.jpg',
    previewImgWebp: 'img/content/look-shot.webp',
    previewImgWebp2x: 'img/content/look-shot@2x.webp',
    reviewCount: 11
  },
  {
    id: 19,
    name: 'Pro Look 4',
    vendorCode: 'PL4CD',
    type: 'Цифровая',
    category: 'Видеокамера',
    description: 'Камера для любителей запечатлеть каждый момент жизни! Слот для SD-карты и возможность смотреть свои записи сразу на TV.',
    previewImg: 'img/content/pro-look-4.jpg',
    level: 'Любительский',
    rating: 5,
    price: 34590,
    previewImg2x: 'img/content/pro-look-4@2x.jpg',
    previewImgWebp: 'img/content/pro-look-4.webp',
    previewImgWebp2x: 'img/content/pro-look-4@2x.webp',
    reviewCount: 1
  },
  {
    id: 20,
    name: 'Pro Look 56F',
    vendorCode: 'PL67T56F',
    type: 'Цифровая',
    category: 'Видеокамера',
    description: '6-кратный зум, чтобы выделить важные подробности. Простые настройки, ЖК-экран, лёгко помещается в руке - всегда можно брать с собой, Li-Ion аккамулятор.',
    previewImg: 'img/content/pro-look-56f.jpg',
    level: 'Любительский',
    rating: 3,
    price: 27990,
    previewImg2x: 'img/content/pro-look-56f@2x.jpg',
    previewImgWebp: 'img/content/pro-look-56f.webp',
    previewImgWebp2x: 'img/content/pro-look-56f@2x.webp',
    reviewCount: 11
  },
  {
    id: 26,
    name: 'GQ Lite',
    vendorCode: 'GO89L',
    type: 'Цифровая',
    category: 'Видеокамера',
    description: 'Небольшая, лёгкая камера с функционалом для начинающего режиссёра. Многократный зум, отличная матрица, усовершенствованный стабилизатор, возможность съёмки по принципц го-про. Для данной камеры можно приобрести водоатталкивающий чехол. а в комплекте штатив. ',
    previewImg: 'img/content/gq-lite.jpg',
    level: 'Любительский',
    rating: 5,
    price: 71900,
    previewImg2x: 'img/content/gq-lite@2x.jpg',
    previewImgWebp: 'img/content/gq-lite.webp',
    previewImgWebp2x: 'img/content/gq-lite@2x.webp',
    reviewCount: 9
  },
  {
    id: 27,
    name: 'Life Pro',
    vendorCode: 'PH67F9R',
    type: 'Коллекционная',
    category: 'Фотоаппарат',
    description: 'Фотокамера премиум класса, позволяет зависывать видео в качестве 4К, обладет 53-кратным зумом, поворотный ЖК-экран, улучшенная матрица, мощный процессор. Линейка выпущена в ограниченном количестве, даже без дополнительного объектива позволит создавать великолепный фотографии за один миг.',
    previewImg: 'img/content/life-pro.jpg',
    level: 'Профессиональный',
    rating: 5,
    price: 199000,
    previewImg2x: 'img/content/life-pro@2x.jpg',
    previewImgWebp: 'img/content/life-pro.webp',
    previewImgWebp2x: 'img/content/life-pro@2x.webp',
    reviewCount: 17
  }
];
