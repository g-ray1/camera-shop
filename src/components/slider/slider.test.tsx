import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { cameras } from '../../mocks/cameras';
import Slider from './slider';

jest.mock('react-redux');

describe('Slider component', () => {
  it('Snapshot', () => {
    const view = render(<Slider products={cameras} />, {wrapper: BrowserRouter});

    expect(view).toMatchSnapshot();
  });

  it('Should return empty element', () => {
    const view = render(<Slider products={[]} />, {wrapper: BrowserRouter});

    expect(view.container).toBeEmptyDOMElement();
  });
});
