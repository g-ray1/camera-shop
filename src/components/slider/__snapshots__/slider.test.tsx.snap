// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page-content__section"
      >
        <section
          class="product-similar"
        >
          <div
            class="container"
          >
            <h2
              class="title title--h3"
            >
              Похожие товары
            </h2>
            <div
              class="product-similar__slider"
            >
              <div
                class="product-similar__slider-list"
              >
                <div
                  class="product-card is-active"
                >
                  <div
                    class="product-card__img"
                  >
                    <picture>
                      <source
                        srcset="/img/content/das-auge.webp, /img/content/das-auge@2x.webp 2x"
                        type="image/webp"
                      />
                      <img
                        alt="Ретрокамера Dus Auge lV"
                        height="240"
                        src="/img/content/das-auge.jpg"
                        srcset="/img/content/das-auge@2x.jpg 2x"
                        width="280"
                      />
                    </picture>
                  </div>
                  <div
                    class="product-card__info"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <div
                        class="rate product-card__rate"
                      >
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-star"
                          />
                        </svg>
                        <p
                          class="visually-hidden"
                        >
                          Рейтинг:
                        </p>
                        <p
                          class="rate__count"
                        >
                          <span
                            class="visually-hidden"
                          >
                            Всего оценок:
                          </span>
                          17
                        </p>
                      </div>
                    </div>
                    <p
                      class="product-card__title"
                    >
                      Ретрокамера Dus Auge lV
                    </p>
                    <p
                      class="product-card__price"
                    >
                      <span
                        class="visually-hidden"
                      >
                        Цена:
                      </span>
                      73450 ₽
                    </p>
                  </div>
                  <div
                    class="product-card__buttons"
                  >
                    <button
                      class="btn btn--purple product-card__btn"
                      type="button"
                    >
                      Купить
                    </button>
                    <a
                      class="btn btn--transparent"
                      href="/product/1"
                    >
                      Подробнее
                    </a>
                  </div>
                </div>
                <div
                  class="product-card is-active"
                >
                  <div
                    class="product-card__img"
                  >
                    <picture>
                      <source
                        srcset="/img/content/fast-shot.webp, /img/content/fast-shot@2x.webp 2x"
                        type="image/webp"
                      />
                      <img
                        alt="FastShot MR-5"
                        height="240"
                        src="/img/content/fast-shot.jpg"
                        srcset="/img/content/fast-shot@2x.jpg 2x"
                        width="280"
                      />
                    </picture>
                  </div>
                  <div
                    class="product-card__info"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <div
                        class="rate product-card__rate"
                      >
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-star"
                          />
                        </svg>
                        <p
                          class="visually-hidden"
                        >
                          Рейтинг:
                        </p>
                        <p
                          class="rate__count"
                        >
                          <span
                            class="visually-hidden"
                          >
                            Всего оценок:
                          </span>
                          24
                        </p>
                      </div>
                    </div>
                    <p
                      class="product-card__title"
                    >
                      FastShot MR-5
                    </p>
                    <p
                      class="product-card__price"
                    >
                      <span
                        class="visually-hidden"
                      >
                        Цена:
                      </span>
                      18970 ₽
                    </p>
                  </div>
                  <div
                    class="product-card__buttons"
                  >
                    <button
                      class="btn btn--purple product-card__btn"
                      type="button"
                    >
                      Купить
                    </button>
                    <a
                      class="btn btn--transparent"
                      href="/product/2"
                    >
                      Подробнее
                    </a>
                  </div>
                </div>
                <div
                  class="product-card is-active"
                >
                  <div
                    class="product-card__img"
                  >
                    <picture>
                      <source
                        srcset="/img/content/instaprinter.webp, /img/content/instaprinter@2x.webp 2x"
                        type="image/webp"
                      />
                      <img
                        alt="Instaprinter P2"
                        height="240"
                        src="/img/content/instaprinter.jpg"
                        srcset="/img/content/instaprinter@2x.jpg 2x"
                        width="280"
                      />
                    </picture>
                  </div>
                  <div
                    class="product-card__info"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <div
                        class="rate product-card__rate"
                      >
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          height="16"
                          width="17"
                        >
                          <use
                            xlink:href="#icon-full-star"
                          />
                        </svg>
                        <p
                          class="visually-hidden"
                        >
                          Рейтинг:
                        </p>
                        <p
                          class="rate__count"
                        >
                          <span
                            class="visually-hidden"
                          >
                            Всего оценок:
                          </span>
                          12
                        </p>
                      </div>
                    </div>
                    <p
                      class="product-card__title"
                    >
                      Instaprinter P2
                    </p>
                    <p
                      class="product-card__price"
                    >
                      <span
                        class="visually-hidden"
                      >
                        Цена:
                      </span>
                      8430 ₽
                    </p>
                  </div>
                  <div
                    class="product-card__buttons"
                  >
                    <button
                      class="btn btn--purple product-card__btn"
                      type="button"
                    >
                      Купить
                    </button>
                    <a
                      class="btn btn--transparent"
                      href="/product/3"
                    >
                      Подробнее
                    </a>
                  </div>
                </div>
                <button
                  aria-label="Предыдущий слайд"
                  class="slider-controls slider-controls--prev"
                  disabled=""
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    height="12"
                    width="7"
                  >
                    <use
                      xlink:href="#icon-arrow"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Следующий слайд"
                  class="slider-controls slider-controls--next"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    height="12"
                    width="7"
                  >
                    <use
                      xlink:href="#icon-arrow"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page-content__section"
    >
      <section
        class="product-similar"
      >
        <div
          class="container"
        >
          <h2
            class="title title--h3"
          >
            Похожие товары
          </h2>
          <div
            class="product-similar__slider"
          >
            <div
              class="product-similar__slider-list"
            >
              <div
                class="product-card is-active"
              >
                <div
                  class="product-card__img"
                >
                  <picture>
                    <source
                      srcset="/img/content/das-auge.webp, /img/content/das-auge@2x.webp 2x"
                      type="image/webp"
                    />
                    <img
                      alt="Ретрокамера Dus Auge lV"
                      height="240"
                      src="/img/content/das-auge.jpg"
                      srcset="/img/content/das-auge@2x.jpg 2x"
                      width="280"
                    />
                  </picture>
                </div>
                <div
                  class="product-card__info"
                >
                  <div
                    class="rate product-card__rate"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <p
                        class="visually-hidden"
                      >
                        Рейтинг:
                      </p>
                      <p
                        class="rate__count"
                      >
                        <span
                          class="visually-hidden"
                        >
                          Всего оценок:
                        </span>
                        17
                      </p>
                    </div>
                  </div>
                  <p
                    class="product-card__title"
                  >
                    Ретрокамера Dus Auge lV
                  </p>
                  <p
                    class="product-card__price"
                  >
                    <span
                      class="visually-hidden"
                    >
                      Цена:
                    </span>
                    73450 ₽
                  </p>
                </div>
                <div
                  class="product-card__buttons"
                >
                  <button
                    class="btn btn--purple product-card__btn"
                    type="button"
                  >
                    Купить
                  </button>
                  <a
                    class="btn btn--transparent"
                    href="/product/1"
                  >
                    Подробнее
                  </a>
                </div>
              </div>
              <div
                class="product-card is-active"
              >
                <div
                  class="product-card__img"
                >
                  <picture>
                    <source
                      srcset="/img/content/fast-shot.webp, /img/content/fast-shot@2x.webp 2x"
                      type="image/webp"
                    />
                    <img
                      alt="FastShot MR-5"
                      height="240"
                      src="/img/content/fast-shot.jpg"
                      srcset="/img/content/fast-shot@2x.jpg 2x"
                      width="280"
                    />
                  </picture>
                </div>
                <div
                  class="product-card__info"
                >
                  <div
                    class="rate product-card__rate"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-star"
                        />
                      </svg>
                      <p
                        class="visually-hidden"
                      >
                        Рейтинг:
                      </p>
                      <p
                        class="rate__count"
                      >
                        <span
                          class="visually-hidden"
                        >
                          Всего оценок:
                        </span>
                        24
                      </p>
                    </div>
                  </div>
                  <p
                    class="product-card__title"
                  >
                    FastShot MR-5
                  </p>
                  <p
                    class="product-card__price"
                  >
                    <span
                      class="visually-hidden"
                    >
                      Цена:
                    </span>
                    18970 ₽
                  </p>
                </div>
                <div
                  class="product-card__buttons"
                >
                  <button
                    class="btn btn--purple product-card__btn"
                    type="button"
                  >
                    Купить
                  </button>
                  <a
                    class="btn btn--transparent"
                    href="/product/2"
                  >
                    Подробнее
                  </a>
                </div>
              </div>
              <div
                class="product-card is-active"
              >
                <div
                  class="product-card__img"
                >
                  <picture>
                    <source
                      srcset="/img/content/instaprinter.webp, /img/content/instaprinter@2x.webp 2x"
                      type="image/webp"
                    />
                    <img
                      alt="Instaprinter P2"
                      height="240"
                      src="/img/content/instaprinter.jpg"
                      srcset="/img/content/instaprinter@2x.jpg 2x"
                      width="280"
                    />
                  </picture>
                </div>
                <div
                  class="product-card__info"
                >
                  <div
                    class="rate product-card__rate"
                  >
                    <div
                      class="rate product-card__rate"
                    >
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        height="16"
                        width="17"
                      >
                        <use
                          xlink:href="#icon-full-star"
                        />
                      </svg>
                      <p
                        class="visually-hidden"
                      >
                        Рейтинг:
                      </p>
                      <p
                        class="rate__count"
                      >
                        <span
                          class="visually-hidden"
                        >
                          Всего оценок:
                        </span>
                        12
                      </p>
                    </div>
                  </div>
                  <p
                    class="product-card__title"
                  >
                    Instaprinter P2
                  </p>
                  <p
                    class="product-card__price"
                  >
                    <span
                      class="visually-hidden"
                    >
                      Цена:
                    </span>
                    8430 ₽
                  </p>
                </div>
                <div
                  class="product-card__buttons"
                >
                  <button
                    class="btn btn--purple product-card__btn"
                    type="button"
                  >
                    Купить
                  </button>
                  <a
                    class="btn btn--transparent"
                    href="/product/3"
                  >
                    Подробнее
                  </a>
                </div>
              </div>
              <button
                aria-label="Предыдущий слайд"
                class="slider-controls slider-controls--prev"
                disabled=""
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="12"
                  width="7"
                >
                  <use
                    xlink:href="#icon-arrow"
                  />
                </svg>
              </button>
              <button
                aria-label="Следующий слайд"
                class="slider-controls slider-controls--next"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  height="12"
                  width="7"
                >
                  <use
                    xlink:href="#icon-arrow"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
